src/chains/sui/deposit.ts

```diff
@@ -0,0 +1,83 @@
+import { getFullnodeUrl, SuiClient } from "@mysten/sui/client";
+import { Ed25519Keypair } from "@mysten/sui/keypairs/ed25519";
+import { Transaction } from "@mysten/sui/transactions";
+
+import { getAddress } from "../../../utils/getAddress";
+import {
+  chainIds,
+  GAS_BUDGET,
+  getCoin,
+  networks,
+  signAndExecuteTransaction,
+  SUI_GAS_COIN_TYPE,
+  toSmallestUnit,
+} from "../../../utils/sui";
+
+type suiDepositParams = {
+  amount: string;
+  receiver: string;
+  token?: string;
+};
+
+type suiOptions = {
+  chainId: (typeof chainIds)[number];
+  gasLimit?: string;
+  gatewayObject?: string;
+  gatewayPackage?: string;
+  signer: Ed25519Keypair;
+};
+
+export const suiDeposit = async (
+  params: suiDepositParams,
+  options: suiOptions
+) => {
+  const gatewayAddress = getAddress("gateway", Number(options.chainId));
+  if (!gatewayAddress) {
+    throw new Error("Gateway address not found");
+  }
+  const gatewayPackage = options.gatewayPackage || gatewayAddress.split(",")[0];
+  const gatewayObject = options.gatewayObject || gatewayAddress.split(",")[1];
```

Comment:
This type of one-line string manipulation could introduce run-time issues. We could check if the `gatewayAddress` string is an actual string that can be split, if the splitted resulting array has multiple indexes, etc. A quick helper leveraging a zod schema check would do.